Display utility associations (#861)
Esri/arcgis-runtime-samples-ios: arcgis-ios-sdk-samples/Utility Network/Display utility association/README.md

> # Display utility associations

Reviewer: ```suggestion
# Display utility association
```

Create graphics for utility associations in a utility network.

![Display utility associations sample](display-utility-association.png)

## Use case

Visualizing utility associations can help you to better understand trace results and the topology of your utility network. For example, connectivity associations allow you to model connectivity between two junctions that don't have geometric coincidence (are not in the same location); structural attachment associations allow you to model equipment that may be attached to structures; and containment associations allow you to model features contained within other features.

## How to use the sample

Pan and zoom around the map. Observe graphics that show utility associations between junctions.

## How it works

1. Create and load an `AGSUtilityNetwork` with a feature service URL.
2. Add an `AGSFeatureLayer` to the map for every `AGSUtilityNetworkSource` of type `edge` or `junction`.
3. Create an `AGSGraphicsOverlay` for the utility associations.

> 4. When the sample starts and every time the viewpoint changes, do the following steps.

Reviewer 2: ```suggestion
4. When the sample is opened and every time the viewpoint changes, do the following steps.
```

> 5. Get the geometry of the map view's extent using `AGSMapView.currentViewpoint(with:).targetGeometry.extent`.

Reviewer 2: This isn't quite right. `AGSMapView.currentViewpoint(with:)` is a symbol name, but `.target.extent` is code. Please either make the whole thing a code expression or rewrite to something like "Get the extent of the map view's target geometry by using `AGSMapView.currentViewpoint(with:)`."

> 6. Get the associations that are within the current extent using `associations(withExtent:completion:)`.

Reviewer 2: ```suggestion
6. Get the associations that are within the current extent using `AGSUtilityNetwork.associations(withExtent:completion:)`.
```
